src/plugins/intel_gpu/src/kernel_selector/cl_kernels/resample_pil_ref.cl

```diff
@@ -207,7 +210,19 @@ KERNEL (resample_vertical_gpu_ref)(  __global INPUT0_TYPE* input
 #elif X_IS_VERTICAL_AXIS == 1
         int in_idx = INTERMEDIATE_BUF_GET_INDEX(b, f, y, vertical_dim + vertical_min);
 #endif
-        ss += input[in_idx] * k[vertical_dim];
+    ss += input[in_idx] * k[vertical_dim];
+    int out_idx = OUTPUT_GET_INDEX(b, f, y, x);
+    if (sizeof(INPUT0_TYPE) == 1) {
+    // For the case of uint8 or int8 where both ENABLE_HORIZONTAL_PASS and ENABLE_VERTICAL_PASS are true,
+    //     - INPUT0_TYPE : uchar or char
+    //     - OUTPUT_TYPE : explicitly float, but actually uchar or char, e.g. 1.0, 2.0, and so on.
+    //     - VERTICAL_INPUT_TYPE : float
+    // At this point type-casting discards the decimal point, which causes a difference from what's expected.
+    // So it needs to round the decimal point and do type-casting in order to reduce the loss.
+        output[out_idx] = FUNC_CALL(round)(ss);
```

Comment:
maybe I'm not understanding the issue, but is it necessary to do the round? it is just a pure loss of data.